triggers/daml/Daml/Trigger.daml

```diff
@@ -257,14 +275,15 @@ runTrigger userTrigger = LowLevel.Trigger
                 runRule userTrigger.rule
         MTransaction transaction -> do
           let acs = applyTransaction transaction state.acs
-              userState = utUpdateState acs (MTransaction transaction) state.userState
+              -- again, we use the commands-in-flight and ACS before the update below
+              userState = utUpdateState state.commandsInFlight acs (MTransaction transaction) state.userState
```

Comment:
This choice of ACS preserves the behavior prior to #7632. With respect to the commands-in-flight, which is new, it seems that it should be the choice that is consistent with the ACS. However, that implies that

```haskell
  updateState = \message -> do
    let commandId = case message of
          MTransaction tx -> tx.commandId
          MComplection cmp -> Some cmp.commandId
          MHeartbeat -> None
    cif <- getCommandsInFlight
    pure $ all (\cid -> Map.member cid cif) commandId
    -- ^ may yield True
```

As such we may want slightly different expected behavior for updateState's commands-in-flight and pendingContracts.

Reply:
Fully agree on keeping commands in flight and the ACS in sync. How do you feel about running `updateState` after updating the ACS and commands in flight instead of the behavior we have now? I don’t have a strong preference either way and it’s definitely something for another PR.

Reply:
#7799